### PR TITLE
Op gateway worker

### PR DESCRIPTION
**Pull request by makoto**
This allows you to host l1 gateway and op gateway on Cloudflare worker

### CODE REVIEW

File: evm-gateway/package.json [hunk 1]
```diff
@@ -54,6 +54,8 @@
   },
   "dependencies": {
     "@chainlink/ccip-read-server": "^0.2.1",
-    "ethers": "^6.7.1"
+    "@ensdomains/ccip-read-cf-worker": "^0.0.1",
+    "ethers": "^6.7.1",
+    "typescript": "^5.2.2"
```
Comment (Arachnid) [inline evm-gateway/package.json:59]: This should be a dev dependency.

File: l1-gateway/README.md [hunk 2]
```diff
@@ -8,4 +9,41 @@ target for testing the entire framework end-to-end.
...
+```
+cd ../l1-verifier l1-gateway
+PROVIDER_URL=$PROVIDER_URL TARGET_ADDRESS=$TARGET_ADDRESS yarn remote_test
+```
+
+## Current deployments
+
+## L1 Gateway
+
+- Verifier = [0xA63381212fBf7AA2D0b4a2cAe453f8c361B11d06](https://goerli.etherscan.io/address/0xA63381212fBf7AA2D0b4a2cAe453f8c361B11d06)
```
Comment (Arachnid) [inline l1-gateway/README.md:43]: Should probably title this section with 'Goerli'. And I don't think addresses for TestL1 and TestL2 are needed in the docs.

File: l1-gateway/src/server.ts [hunk 3]
```diff
@@ -10,10 +11,11 @@ const program = new Command()
 program.parse();
 
 const options = program.opts();
-
 const provider = new ethers.JsonRpcProvider(options.providerUrl);
-const gateway = new EVMGateway(new L1ProofService(provider));
-const app = gateway.makeApp('/');
+const proof = new EVMGateway(new L1ProofService(provider));
```
Comment (Arachnid) [inline l1-gateway/src/server.ts:15]:
```suggestion
const gateway = new EVMGateway(new L1ProofService(provider));
```

File: l1-verifier/scripts/remote.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import hre from "hardhat";
```
Comment (Arachnid) [inline l1-verifier/scripts/remote.ts:1]: What is this script for?

File: l1-verifier/test/testL1Verifier.ts [hunk 5]
```diff
@@ -43,10 +44,11 @@ describe('L1Verifier', () => {
     // provider.on("debug", (x: any) => console.log(JSON.stringify(x, undefined, 2)));
     signer = await provider.getSigner(0);
 
-    const server = makeL1Gateway(provider as unknown as JsonRpcProvider);
-    gateway = server.makeApp('/');
-
-    const getUrl = FetchRequest.createGetUrlFunc();
+    const proof = makeL1Gateway(provider as unknown as JsonRpcProvider);
```
Comment (Arachnid) [inline l1-verifier/test/testL1Verifier.ts:47]:
```suggestion
    const gateway = makeL1Gateway(provider as unknown as JsonRpcProvider);
```

File: l1-verifier/test/testL1Verifier.ts [hunk 6]
```diff
@@ -43,10 +44,11 @@ describe('L1Verifier', () => {
...
+    const server = new Server()
+    proof.add(server)
+    const gateway = server.makeApp('/')
```
Comment (Arachnid) [inline l1-verifier/test/testL1Verifier.ts:50]:
```suggestion
    const app = server.makeApp('/')
```

File: op-gateway/src/server.ts [hunk 7]
```diff
@@ -30,15 +31,17 @@ program.parse();
   const l1Provider = new JsonRpcProvider(options.l1ProviderUrl);
   const l2Provider = new JsonRpcProvider(options.l2ProviderUrl);
 
-  const gateway = new EVMGateway(
+  const proof = new EVMGateway(
```
Comment (Arachnid) [inline op-gateway/src/server.ts:34]:
```suggestion
  const gateway = new EVMGateway(
```

File: op-gateway/src/worker.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    L1_PROVIDER_URL: string
+    L2_PROVIDER_URL: string
+    L2_OUTPUT_ORACLE: string
+    DELAY: number
+}
+interface Router{
+  handle:(request:Request)=> void
+}
+let app:Router  
+async function fetch(request:Request, env:Env){
```
Comment (Arachnid) [inline op-gateway/src/worker.ts:12]: Could we provide some kind of utility function on `evm-gateway` for this, to reduce the boilerplate?

File: op-verifier/deploy_l1/00_op_verifier.ts [hunk 9]
```diff
@@ -3,16 +3,22 @@ import {DeployFunction} from 'hardhat-deploy/types';
...
-  const {deployments, getNamedAccounts} = hre;
+  const {deployments, getNamedAccounts, network} = hre;
   const {deploy} = deployments;
-
   const {deployer} = await getNamedAccounts();
-
-  const opAddresses = await (await fetch("http://localhost:8080/addresses.json")).json();
-
+  let L2_OUTPUT_ORACLE_ADDRESS, GATEWAY_URL
+  if(network.name === 'opDevnetL1'){
```
Comment (Arachnid) [inline op-verifier/deploy_l1/00_op_verifier.ts:10]: Can you use a mapping of URLs instead of an if statement here?

**Review by Arachnid — CHANGES_REQUESTED** [review]

File: l1-gateway/README.md (hunk 2 above)
Comment (makoto) [reply l1-gateway/README.md:43]: I will comment as Example, as we also have an instruction of how to test that requires to know l1 test address

File: l1-verifier/scripts/remote.ts (hunk 4 above)
Comment (makoto) [reply l1-verifier/scripts/remote.ts:1]: It's mentioned here https://github.com/ensdomains/evmgateway/pull/4/files#diff-a1176ce8a8e70ac565ce2aa2e27bc770a2ac7b36e803e985bf2dd9419b634c64R34

File: op-verifier/deploy_l1/00_op_verifier.ts (hunk 9 above)
Comment (makoto) [reply op-verifier/deploy_l1/00_op_verifier.ts:19]: I still need if statement to fetch L2_OUTPUT_ORACLE_ADDRESS address dynamically though

**Review by Arachnid — APPROVED** [review]

**Pull request merged**